homeassistant/components/forked_daapd/media_player.py

```diff
@@ -668,69 +668,68 @@ async def async_play_media(
         if media_type == MediaType.MUSIC:
             media_id = async_process_play_media_url(self.hass, media_id)
 
-            saved_state = self.state  # save play state
-            saved_mute = self.is_volume_muted
-            sleep_future = asyncio.create_task(
-                asyncio.sleep(self._tts_pause_time)
-            )  # start timing now, but not exact because of fd buffer + tts latency
-            await self._pause_and_wait_for_callback()
-            await self._save_and_set_tts_volumes()
-            # save position
-            saved_song_position = self._player["item_progress_ms"]
-            saved_queue = (
-                self._queue if self._queue["count"] > 0 else None
-            )  # stash queue
+            await self._async_announce(media_id)
+        else:
+            _LOGGER.debug("Media type '%s' not supported", media_type)
+
+    async def _async_announce(self, media_id: str):
+        saved_state = self.state  # save play state
+        saved_mute = self.is_volume_muted
+        sleep_future = asyncio.create_task(
+            asyncio.sleep(self._tts_pause_time)
+        )  # start timing now, but not exact because of fd buffer + tts latency
+        await self._pause_and_wait_for_callback()
+        await self._save_and_set_tts_volumes()
+        # save position
+        saved_song_position = self._player["item_progress_ms"]
+        saved_queue = self._queue if self._queue["count"] > 0 else None  # stash queue
+        if saved_queue:
+            saved_queue_position = next(
+                i
+                for i, item in enumerate(saved_queue["items"])
+                if item["id"] == self._player["item_id"]
+            )
+        self._tts_requested = True
+        await sleep_future
+        await self._api.add_to_queue(uris=media_id, playback="start", clear=True)
+        try:
+            await asyncio.wait_for(self._tts_playing_event.wait(), timeout=TTS_TIMEOUT)
```

Comment:
Side note: We prefer using `async_timeout.timeout`.

Reply:
Thanks, will incorporate this.

Reply:
See #78451